### ISSUE TITLE
prg injection is unreliable

### ISSUE THREAD

**Issue by rittwage** [mist-devel/c64#11]
Certain PRG files do not run properly after injection.

Example: swinth.prg, fort apocalypse.prg

**Comment by gyurco** [comment]
Well, fully emulating the LOAD command can be tricky. Even vice chooses a different way - creates a virtual D64 and loads the PRG with the BASIC from it.

**Comment by gyurco** [comment]
...however try this, might improve the situation a bit.
[C64_mist-230331.zip](https://github.com/mist-devel/c64/files/11125066/C64_mist-230331.zip)

**Comment by rittwage** [comment]
Yes, this is better. This fixed swinth.prg. FortApocalypse.prg could just be an issue with my file.

**Comment by rittwage** [comment]
[Fort Apocalypse.zip](https://github.com/mist-devel/c64/files/11125197/Fort.Apocalypse.zip)

This goes crazy on the core. Works in VICE and on real C64.

**Comment by gyurco** [comment]
For me, it's the same with Vice and MiST - an upper HUD appears, then nothing happens.

**Comment by rittwage** [comment]
Hmm... I downloaded VICE 3.7 (windows x64) and drag/drop it in and it runs for me normally.

**Comment by gyurco** [comment]
Then it must be some unusal setting.

**Comment by gyurco** [comment]
BTW, I extracted the PRG from https://csdb.dk/release/?id=19268, and it works on MiST.
[fort apocalypse.zip](https://github.com/mist-devel/c64/files/11127075/fort.apocalypse.zip)

**Comment by rittwage** [comment]
Yes, that Fort Apocalypse PRG is a bad example. The game does normally work, just that version acts funny. I have verified the same thing happens when run in D64, so it's not the PRG injection causing any issue with it. Also, the same thing happens in the MiSTer core, so it's not anything with MiST either.  If I run into any others with this new binary, I will report it.
